client/layout/navigation.js

```diff
@@ -6,18 +6,32 @@ import {
 	WooNavigationItem,
 	getNewPath,
 	getPersistedQuery,
+	getQueryExcludedScreens,
 } from '@woocommerce/navigation';
 import { Link } from '@woocommerce/components';
 import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import getReports from '../analytics/report/get-reports';
 import { getPages } from './controller';
 import { isWCAdmin } from '../dashboard/utils';
+import { addHistoryListener } from '../navigation/utils';
 
 const NavigationPlugin = () => {
+	const [ persistedQuery, setPersistedQuery ] = useState(
+		getPersistedQuery()
+	);
+
+	// Update the persisted queries when history is updated
+	useEffect( () => {
+		return addHistoryListener( () => {
+			setTimeout( () => setPersistedQuery( getPersistedQuery() ), 0 );
+		} );
```

Comment:
In testing I see that going to `Customers` and then back to a report looses the persisted query. Do you think we could skip this function call to `setPersistedQuery` when on the Customers page or another excluded page?

Reply:
Great call @psealock . Just committed this change, and ended up refactoring a few things as well.